include/envoy/network/transport_socket.h

```diff
@@ -100,6 +104,11 @@ class TransportSocket {
    */
   virtual void closeSocket(Network::ConnectionEvent event) PURE;
 
+  /**
+   * Half-closes the transport socket. For TCP, this sends a FIN.
+   */
+  virtual void halfCloseSocket() PURE;
```

Comment:
I started with including it in closeSocket(), but gave up because it shared no code with the other close paths, and the parameter type was wrong.  It would have required adding a new enum value to ConnectionEvent that only made sense in this context, not in raising events to filters.

Reply:
It might be possible to merge those two methods. I don't think the parameter of `closeSocket` is actually used.